server/cmt_cmds.go

```diff
@@ -115,3 +122,130 @@ func VersionCmd() *cobra.Command {
 		},
 	}
 }
+
+// QueryBlocksCmd returns a command to search through blocks by events.
+func QueryBlocksCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "blocks",
+		Short: "Query for paginated blocks that match a set of events",
+		Long: `Search for blocks that match the exact given events where results are paginated.
+The events query is directly passed to CometBFT's RPC BlockSearch method and must
+conform to CometBFT's query syntax.
+Please refer to each module's documentation for the full set of events to query
+for. Each module documents its respective events under 'xx_events.md'.
+`,
+		Example: fmt.Sprintf(
```

Comment:
Maybe I have done something wrong, but this is not working (simply using the example):     https://asciinema.org/a/YknCh0VUhOdPDe0XVEu97QtjQ

Reply:
Idk what's happening with it, but in the cometbft rpc docs it says it only works on begin/end block events - https://github.com/cometbft/cometbft/blob/af13a8a639aa8d66e09b8f348b010e15f8fa85d6/rpc/openapi/openapi.yaml#L1148

Reply:
```
# get the latest block_results
curl localhost:26657/block_results
            ...
               {
                "type": "message", 
                "attributes": [
                    {
                        "key": "sender", 
                        "value": "cosmos17xpfvakm2amg962yls6f84z3kell8c5lserqta", 
                        "index": true
                    }
                ]
            }
        ...
# choose a BeginBlock Event
 ./build/simd q blocks --query "message.sender = 'cosmos17xpfvakm2amg962yls6f84z3kell8c5lserqta'" --page 1 --limit 30

...
    round: 0
    signatures:
    - block_id_flag: BLOCK_ID_FLAG_COMMIT
      signature: oj5qsGjvNwc+76tksuIaTjNEE0wzORjRt/JvOYrofbui7VZqXTqhbgKpIqp+zghxyc71RsCEh3fTaCDiNc35DA==
      timestamp: "2023-02-14T05:14:11.370953678Z"
      validator_address: 7tCr6IunraLy6Ikv4FQlp9/Uh3g=
count: "30"
limit: "30"
page_number: "1"
page_total: "922"
total_count: "27655"
...
```

Reply:
@julienrbrt - Am I missing something here?

Reply:
Nope, all good, my bad. Could you please implement the other feedback and then it's good to go.

Reply:
ok, cool thanks